dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -2422,9 +2421,7 @@ RtpsUdpDataLink::RtpsWriter::gather_gaps_i(const RepoId& reader,
     bitmap.length(bitmap_num_longs(base, gaps.high()));
     (void) gaps.to_bitmap(bitmap.get_buffer(), bitmap.length(), num_bits);
   } else {
-    bitmap.length(1);
     bitmap[0] = 0;
```

Comment:
Now that the length is zero (default) we can't index the first position (zero) or we'll access invalid memory. Same for linke 1833.